### PR TITLE
skeditor: init at 2.8.1

### PR DESCRIPTION
**Pull request by Eveeifyeve**
This pr contains a new application to nixpkgs called [SKEditor](https://github.com/SkEditorTeam/SkEditor) It's a special Editor for [Skript](https://github.com/SkriptLang/Skript) users.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/sk/skeditor/package.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  dotnet-runtime = dotnetCorePackages.runtime_8_0;
+
+  passthru.updateScript = ./update.sh;
+
+  meta = with lib; {
+    description = "SkEditor is an app for editing Skript files";
+    homepage = "https://github.com/SkEditorTeam/SkEditor";
+    changelog = "https://github.com/SkEditorTeam/SkEditor/releases/tag/v${version}";
+    licence = licenses.mit;
+    maintainers = with maintainers; [ eveeifyeve ];
```
Comment (Aleksanaa) [inline pkgs/by-name/sk/skeditor/package.nix:32]:
```suggestion
  meta = {
    description = "SkEditor is an app for editing Skript files";
    homepage = "https://github.com/SkEditorTeam/SkEditor";
    changelog = "https://github.com/SkEditorTeam/SkEditor/releases/tag/v${version}";
    licence = lib.licenses.mit;
    maintainers = with lib.maintainers; [ eveeifyeve ];
```

File: pkgs/by-name/sk/skeditor/package.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  executables = [ "SkEditor" ];
+  nugetDeps = ./nuget-deps.nix;
+
+  dotnet-sdk = dotnetCorePackages.sdk_8_0;
+  dotnet-runtime = dotnetCorePackages.runtime_8_0;
+
+  passthru.updateScript = ./update.sh;
+
+  meta = with lib; {
+    description = "SkEditor is an app for editing Skript files";
```
Comment (Aleksanaa) [inline pkgs/by-name/sk/skeditor/package.nix:28]:
```suggestion
    description = "App for editing Skript files";
```

File: pkgs/by-name/sk/skeditor/package.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  fetchFromGitHub,
+  dotnetCorePackages,
+}:
+buildDotnetModule rec {
+  pname = "skeditor";
+  version = "2.8.1";
+
+  src = fetchFromGitHub {
+    owner = "skeditorteam";
+    repo = pname;
```
Comment (Aleksanaa) [inline pkgs/by-name/sk/skeditor/package.nix:13]:
```suggestion
    repo = "skeditor";
```

File: pkgs/by-name/sk/skeditor/package.nix (hunk 1 above)
Comment (Eveeifyeve) [reply pkgs/by-name/sk/skeditor/package.nix]: You need to update this suggestion @Aleksanaa,

Comment (Aleksanaa) [reply pkgs/by-name/sk/skeditor/package.nix]: > You need to update this suggestion @Aleksanaa,

We don't work this way. At the end you still have to squash in your local branch.

File: pkgs/by-name/sk/skeditor/package.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  dotnet-sdk = dotnetCorePackages.sdk_8_0;
+  dotnet-runtime = dotnetCorePackages.runtime_8_0;
+
+  passthru.updateScript = ./update.sh;
+
+  meta = {
+    description = "SkEditor is an app for editing Skript files";
+    homepage = "https://github.com/SkEditorTeam/SkEditor";
+    changelog = "https://github.com/SkEditorTeam/SkEditor/releases/tag/v${version}";
+    licence = lib.licenses.mit;
```
[the remark this answers is not in the document]
Comment (Eveeifyeve) [reply pkgs/by-name/sk/skeditor/package.nix:31]: Yeah I hadn't fix that eval error but yeah I will goahead and add this then squash this.

File: pkgs/by-name/sk/skeditor/package.nix (hunk 2 above)
Comment (Eveeifyeve) [reply pkgs/by-name/sk/skeditor/package.nix:28]: I would reword app to be "Skript Code Editor for Skript" but I just pulled that description but I guess that works for now.

Comment (Eveeifyeve) [reply pkgs/by-name/sk/skeditor/package.nix:28]: Or even better "An IDE for Skript".
